### PR TITLE
MM-28263: Cypress/E2E: Automate backlogs - Messaging > Pinned Posts part 2

### PR DESCRIPTION
**Pull request by agnivade**
This PR automates

- MM-T2173
- MM-T2174
- MM-T2175
- MM-T2177
- MM-T2922

https://mattermost.atlassian.net/browse/MM-28263

### CODE REVIEW

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:

File: e2e/cypress/integration/messaging/pinned_posts_part2_spec.js [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+    it('MM-T2175 Pin a post in a DM channel Pin a post while viewing empty pinned post RHS', () => {
+        // # Create DM Channel
+        cy.apiCreateDirectChannel([testUser.id, otherUser.id]).then(() => {
+            cy.apiLogin(testUser);
+
+            // # Visit the DM channel
+            cy.visit(`/${testTeam.name}/channels/${testUser.id}__${otherUser.id}`);
+
+            // # Click pin icon
+            cy.get('#channelHeaderPinButton').should('be.visible').click();
```
Comment (saturninoabril) [inline e2e/cypress/integration/messaging/pinned_posts_part2_spec.js:110]: Failing intermittently on me. After visiting a page via URL, good practice is to add timeout when getting the first element.
```javascript
cy.get('#channelHeaderPinButton', {timeout: TIMEOUTS.ONE_MIN}).should('be.visible').click();
```

File: e2e/cypress/integration/messaging/pinned_posts_part2_spec.js [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+            cy.postMessage('Hello');
+
+            return cy.getLastPostId();
+        }).then((postId) => {
+            // # Pin the post
+            pinPost(postId);
+
+            // # Now open RHS
+            cy.get(`#post_${postId}`).trigger('mouseover');
+            cy.get(`#CENTER_commentIcon_${postId}`).click({force: true});
```
Comment (saturninoabril) [inline e2e/cypress/integration/messaging/pinned_posts_part2_spec.js:122]: I thought the requirement is not to open the thread but to observe "Pinned posts" in RHS which will eventually  change to contain the recently pinned post.

File: e2e/cypress/integration/messaging/pinned_posts_part2_spec.js [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+        // # Post a message
+        cy.postMessage('Hello again');
+
+        cy.getLastPostId().then((postId) => {
+            // # Pin the post
+            pinPost(postId);
+
+            // # Now open RHS
+            cy.get(`#post_${postId}`).trigger('mouseover');
+            cy.get(`#CENTER_commentIcon_${postId}`).click({force: true});
```
Comment (saturninoabril) [inline e2e/cypress/integration/messaging/pinned_posts_part2_spec.js:152]: Same here. It should observe the the change to "Pinned posts" in RHS.

File: e2e/cypress/integration/messaging/pinned_posts_part2_spec.js [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+
+            // # Click pin icon
+            cy.get('#channelHeaderPinButton').should('be.visible').click();
+
+            // * Should not have any pinned posts
+            cy.get('#search-items-container .no-results__title').should('have.text', 'No pinned posts yet');
+
+            // * Post should be deleted
+            cy.get(`#post_${postId}`).should('not.exist');
+            cy.get(`#rhsPost_${postId}`).should('not.exist');
```
Comment (saturninoabril) [inline e2e/cypress/integration/messaging/pinned_posts_part2_spec.js:171]: I don't see where "Note: If another user deleted the message, message displays "(message deleted)". Pinned badge still shows, but all disappear on refresh" is fulfilled.

**Review by saturninoabril — CHANGES_REQUESTED** [review]
Thanks @agnivade! Pease see comments.

File: e2e/cypress/integration/messaging/pinned_posts_part2_spec.js (hunk 2 above)
Comment (agnivade) [reply e2e/cypress/integration/messaging/pinned_posts_part2_spec.js:122]: Oh yeah that's right. Thanks.

File: e2e/cypress/integration/messaging/pinned_posts_part2_spec.js (hunk 4 above)
Comment (agnivade) [reply e2e/cypress/integration/messaging/pinned_posts_part2_spec.js:171]: I thought it was a note and not a compulsory step to do. Will look into it. Thanks.

Comment (saturninoabril) [reply e2e/cypress/integration/messaging/pinned_posts_part2_spec.js]: Yup, you're right. I'm just not sure if it's covered somewhere else. Thanks for adding here!

**Review by saturninoabril — APPROVED** [review]
@agnivade Thanks! LGTM, tested and all passed.

File: e2e/cypress/integration/messaging/pinned_posts_part2_spec.js [hunk 5]
```diff
@@ -0,0 +1,301 @@
...
+            // * Should not have any pinned posts
+            cy.get('#search-items-container .no-results__title').should('have.text', 'No pinned posts yet');
+        });
+    });
+
+    it('MM-T2174 Switching channels in center also changes pinned posts RHS', () => {
+        // # Login
+        cy.apiLogin(testUser);
+
+        // # Visit a test channel
```
Comment (lindy65) [inline e2e/cypress/integration/messaging/pinned_posts_part2_spec.js:74]: Hi @agnivade - do you need to add `and post a message` to this line and to the code below?

Comment (agnivade) [reply e2e/cypress/integration/messaging/pinned_posts_part2_spec.js:74]: The post already exists in the channel. So we don't want to add a new one. Getting the last post is sufficient.

Comment (lindy65) [reply e2e/cypress/integration/messaging/pinned_posts_part2_spec.js:74]: Ah, ok... For every other test in the PR you have code to make a post first and it was missing on this one so I just wondered.  Also, will these tests run concurrently in Cypress or does each test run individually from start to finish? (Just trying to understand how we re-use a post from a previous test case why the tests are running)

Comment (agnivade) [reply e2e/cypress/integration/messaging/pinned_posts_part2_spec.js:74]: They run individually from start to finish. It's exactly like how you would manually run the tests by hand one by one.

**Review by lindy65 — APPROVED** [review]
Thanks @agnivade - looks good to me 👍

**Pull request merged**